client-ts/signalr/src/HttpConnection.ts

```diff
@@ -87,41 +92,61 @@ export class HttpConnection implements IConnection {
                     headers.set("Authorization", `Bearer ${token}`);
                 }
 
-                const negotiatePayload = await this.httpClient.post(this.resolveNegotiateUrl(this.baseUrl), {
-                    content: "",
-                    headers,
-                });
+                await this.setNegotiationResponse(headers);
 
-                const negotiateResponse: INegotiateResponse = JSON.parse(negotiatePayload.content as string);
-                this.connectionId = negotiateResponse.connectionId;
+                this.connectionId = this.negotiateResponse.connectionId;
 
                 // the user tries to stop the the connection when it is being started
                 if (this.connectionState === ConnectionState.Disconnected) {
                     return;
                 }
 
-                if (this.connectionId) {
-                    this.url = this.baseUrl + (this.baseUrl.indexOf("?") === -1 ? "?" : "&") + `id=${this.connectionId}`;
-                    this.transport = this.createTransport(this.options.transport, negotiateResponse.availableTransports, transferFormat);
+                for (const transport of this.allTransports) {
```

Comment:
Should be doing something more similar to https://github.com/aspnet/SignalR/blob/dev/src/Microsoft.AspNetCore.Sockets.Client.Http/HttpConnection.cs#L228-L234

Reply:
Yes, why do we hardcode the list?

Reply:
`createTransport` and `resolveTransport` already does this looping. We should move this logic into `createTransport`. That code is already using `resolveTransport` to determine if the client can support the transport, all we should need is to move the `try { connect(); break; } catch { /* continue */ }` logic into the loop in `createTransport`